diffcp/cone_program.py

```diff
@@ -131,31 +139,44 @@ def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
     m, n = A.shape
     N = m + n + 1
     cones = cone_lib.parse_cone_dict(cone_dict)
+    cones_parsed = cone_lib.parse_cone_dict_cpp(cones)
     z = (x, y - s, np.array([1]))
     u, v, w = z
+
     Q = sparse.bmat([
         [None, A.T, np.expand_dims(c, - 1)],
         [-A, None, np.expand_dims(b, -1)],
         [-np.expand_dims(c, -1).T, -np.expand_dims(b, -1).T, None]
     ])
-    D_proj_dual_cone = _diffcp.dprojection(
-        v, cone_lib.parse_cone_dict_cpp(cones), True)
+
+    if mode == "dense" or mode == "sparse":
+        D_proj_dual_cone = cone_lib.dpi_explicit(v, cones, dual=True)
+        try:
+            M = (Q - sparse.eye(N)) @ dpi_explicit(z, cones) + sparse.eye(N)
+            MT = M.T
+        except NotImplementedError:
+            print("PSD cone not supported; switching to mode=lsqr.")
```

Comment:
Hmm, I think we can completely get rid of this error by using the PSD linop for now. Adding that in

Reply:
Although actually it may be nice to still keep these around in `cones.py` for debugging/development in case an unimplemented cone is passed in. Want me to update those from `NotImplementedError` to `ValueError` there?